### ISSUE TITLE
BUG: Progressive degradation of the pencil tool performance after prolonged use of Obsidian

### ISSUE THREAD

**Issue by HoodyEz** [zsviczian/obsidian-excalidraw-plugin#2167]
### Have you searched for existing issues (including closed ones)?

- [X] Yes, I have searched the existing issues.

### Does this bug persist in a new vault with only Excalidraw installed?

- [X] Yes, I have verified the issue persists.

### Your environment

SYSTEM INFO:
	Obsidian version: v1.7.7
	Installer version: v1.7.6
	Operating system: Windows 11 Home 10.0.26100
	Login status: logged in
	Language: en
	Catalyst license: none
	Insider build toggle: off
	Live preview: on
	Base theme: dark
	Community theme: none
	Snippets enabled: 6
	Restricted mode: off
	Plugins installed: 17
	Plugins enabled: 16
		1: PDF++ v0.40.13
		2: Calendar v1.5.10
		3: Fleeting Notes Sync v0.6.3
		4: Image Context Menus v1.9.1
		5: Project Browser v0.1.5
		6: Tasks v7.14.0
		7: Projects v1.17.4
		8: Advanced URI v1.44.2
		9: Outliner v4.8.1
		10: Latex Suite v1.9.7
		11: Editing Toolbar v2.4.16
		12: Recent Files v1.7.3
		13: Homepage v4.0.7
		14: Tracker v1.14.0
		15: Heading Level Indent v2.0.0
		16: Excalidraw v2.7.1

### Describe the bug

The pencil tool in Excalidraw becomes less responsive over time, making drawn lines feel increasingly fragmented (like disconnected dots instead of continuous strokes). This occurs after using Obsidian for more than 1 hour.

I've been noticing for a while that the pencil tool loses smoothness the longer I use an Obsidian session. Previously, I used the "Tray" plugin to avoid the loading time when opening Obsidian, but due to the issue I mentioned earlier, I decided to uninstall the plugin in case there was a compatibility issue. However, as I continued using it, I experienced the same problem again.

### Steps to reproduce

1. Open Obsidian with the Excalidraw plugin activated.
2. Draw lines using the pencil tool at the start of the session (note the smoothness).
3. Keep using Obsidian for more than 1 hour (open pdfs, writing notes, using excalidraw plugin)
4. Repeat the same drawing process mentioned in step 2.
5. Observe the delay in the strokes.

### Expected behavior

The pencil tool should maintain consistent smoothness regardless of how long Obsidian remains open.

### Additional context

![image](https://github.com/user-attachments/assets/164ff5f4-de17-41b9-8812-b71fbc067c3e)

[fluidez handwriting.webm](https://github.com/user-attachments/assets/feac2765-8985-46a7-a17b-edd2b598287c)

In the video, the smoothness of the pencil tool is not very evident at first glance, but upon closer inspection, you can notice small skips in each stroke, similar to the drawing above. However, when I close and reopen Obsidian, the pencil tool feels much smoother, and the strokes are more fluid.

**Comment by zsviczian** [comment]
yes, there are known memory leak issues. I hope to get to their bottom one day. Until then, I recommend restarting obsidian when things slow down...

**Comment by HoodyEz** [comment]
Ouu, I understand, thanks anyway.

**Comment by zsviczian** [comment]
so maybe I am too quick to take the blame on the Excalidraw plugin. I have spent a fair number of weeks debugging memory leak and fixing most of the issues.... to the extent that recently when I monitored the situation I could not register any significant problems... however along the way as I was figuring out the issues with Excalidraw I did notice that pretty much all the plugins leak memory. You describe using Obsidian extensively... so it could very well be that it is a different plugin that causes the leak and Excalidraw takes the hit.

Long story short... Obsidian plugins are all hobby projects developed by single individuals, not professional commercial products developed by armies of people. You get more features then with any other tool, and you get it for free... the tradeoff of restarting every now and then does not sound that bad to me... 😉
